core/src/main/java/org/apache/accumulo/core/spi/crypto/AESCryptoService.java

```diff
@@ -132,11 +134,11 @@ public FileEncrypter getFileEncrypter(CryptoEnvironment environment) {
   @Override
   public FileDecrypter getFileDecrypter(CryptoEnvironment environment) {
     CryptoModule cm;
-    byte[] decryptionParams = environment.getDecryptionParams();
-    if (decryptionParams == null || checkNoCrypto(decryptionParams))
+    var decryptionParams = environment.getDecryptionParams();
+    if (decryptionParams.isEmpty() || checkNoCrypto(decryptionParams.get()))
       return new NoFileDecrypter();
 
-    ParsedCryptoParameters parsed = parseCryptoParameters(decryptionParams);
+    ParsedCryptoParameters parsed = parseCryptoParameters(decryptionParams.get());
     Key kek = loadDecryptionKek(parsed);
```

Comment:
From what I can tell, loadDecryptionKek is only called once, and inside of the method it only parses and stores one key for decryption in the variable `decryptingKeys`. However, `decryptingKeys` is a Map, which makes sense because IIRC you will only use 1 current un-expired key for encrypting, but you may need the current and all historical keys (the expired or rotated ones) for decrypting.